client/src/panels/panel_declarations/finances/drr_dp_resources/crso_by_prog.js

```diff
@@ -322,9 +322,23 @@ const get_calculate_func = (is_fte) => {
       ),
     }));
 
-    const relevant_data = is_fte ? fte_data : exp_data;
-    const first_year_top_2_programs = _.chain(relevant_data)
-      .sortBy(({ data }) => _.first(data))
+    const relevant_data = is_fte ? queried_fte.data : queried_exp.data;
+    const most_recent_top_2_programs = _.chain(relevant_data)
+      .flatMap((program_data) => ({
+        prgm: program_data.prgm,
+        value:
+          program_data[is_fte ? "{{pa_last_year}}" : "{{pa_last_year}}exp"],
```

Comment:
I don't have an example in mind, but theoretically a CR will die and then hang around, so `pa_last_year` might not always be the "most recent" year for actual spending.

Fine to hardcode planning to `planning_year_1` because, if it has any planned data, it will exist.

Part of the pain points of putting both in to one graph, it needs to work in a lot of cases:
  1) dead CR with historical actual spending, no planned
  1.5) not sure if this _can_ happen, but might be a non-dead CR with no planned some day :shrug:
  2) ongoing actual spending\*, planned spending\**
  3) planned spending only

\* anywhere from one to five years, but can assume they are consecutive and include the most recent year
\** should always be three years out, should have data for at least planning year 1 but might be dead after that

Reply:
Okay, if I change the hard coded `pa_last_year` to be dynamic, I should be covered for all 3.5 cases I think